shark_turbine/kernel/wave/wave.py

```diff
@@ -298,8 +319,29 @@ def test_execute(self, args, kwargs):
             )
             vm_context = ctx.vm_context
 
-            func = mod.lookup_function("isolated_benchmark")
-            _invoke(vm_context, device, func, kernel_inputs, kernel_outputs)
+            func_name = "isolated_benchmark"
+            if run_bench:
+                benchmark_flags = {}
+                if bench_batch_size is not None:
+                    benchmark_flags["batch_size"] = int(bench_batch_size)
+
+                if bench_repetitions is not None:
+                    benchmark_flags["benchmark_repetitions"] = int(bench_repetitions)
+
+                inputs = [inp.numpy() for inp in kernel_inputs]
+                benchmark_results = bench.benchmark_module(
+                    mod,
+                    entry_function=func_name,
+                    device=device,
+                    inputs=inputs,
+                    **benchmark_flags,
+                )
+                print("Benchmark results:")
```

Comment:
Can we also dump this to a file? Just thinking ahead, when we do tuning or even just a sweep, we would like to know the configuration that produced the benchmark number. Also should we wrap the print with a logger?

Reply:
Thanks!